### PR TITLE
Add: `xs` size to LoadingSpinner

### PR DESCRIPTION
**Pull request by mcwinter07**
## Why
This is part of the prework for the new Button to enable content dense layouts.

## What
- Adds `xs` size to `LoadingSpinner`
- Migrate from SCSS to CSS variables and stylesheet
- Remove width declaration from svg and set it on the container
  - This can now be controlled via the `--loading-spinner-size` variable in the `classNameOverride` if required
- Update docs examples

### CODE REVIEW

File: packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.mdx
Comment (mcwinter07) [inline packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.mdx:45]: I'm a little dubious in terms of how much we want to promote this so curious for thoughts @dougmacknz @caseysinclair. Happy to leave it out of here and just have it as a thing folks can do if they need an escape hatch :)

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx
Comment (mcwinter07) [inline packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx:7]: Anyone think this is worth exporting?

Comment (dougmacknz) [reply packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx:7]: I can't see anyone wanting this so yeah I'd leave it private unless/until someone asks for it

File: packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.mdx
Comment (dougmacknz) [reply packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.mdx:45]: Yeah I'd leave this out

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx
Comment (HeartSquared) [reply packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx:7]: Wouldn't even make a type for it tbh 😅

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx
Comment (HeartSquared) [inline packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx:17]: I'd convert these to individual components instead of a function that returns a JSX.Element

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx
Comment (HeartSquared) [inline packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx:108]: You reckon we could remove this? 👀

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.module.css
Comment (HeartSquared) [inline packages/components/src/Loading/LoadingSpinner/LoadingSpinner.module.css:15]: We should try to keep specificity low where possible, so I'd remove the `.loadingSpinner` part. CSS modules will prevent name clashing.

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.module.css
Comment (HeartSquared) [inline packages/components/src/Loading/LoadingSpinner/LoadingSpinner.module.css:2]: If we expect that size will always exist, we could probably safely move this into a class so it's clear that it's for `md`. 
That being said, I could see the argument for missing a default I guess 🤷🏻‍♀️  I'll let you decide :)

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.module.css
Comment (HeartSquared) [inline packages/components/src/Loading/LoadingSpinner/LoadingSpinner.module.css:10]: Given spinners are closer to images and icons, we should probably actually use `px` values for them as they don't need to scale with increased font size (have been considering that spacing probably makes sense to be too in most cases, but that's a different problem).

File: packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.mdx
Comment (HeartSquared) [reply packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.mdx:45]: Agreed, we should leave this out because we aren't encouraging the use of manipulating the size (unlike new Icon).

File: packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stickersheet.stories.tsx
Comment (HeartSquared) [inline packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stickersheet.stories.tsx:42]: You can leave this one out :)

File: packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx
Comment (HeartSquared) [inline packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx:30]: Curious about the need to increase these examples?

File: packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.mdx
Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.mdx:45]: Cool 👍

File: packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx
Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx:30]: I was aligning this to the examples Jess has in Figma, figured while we probably don't want to introduce a color prop we can at least show the one that match

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.module.css
Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/LoadingSpinner.module.css:10]: Makes sense to me, will make the change 👍

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.module.css
Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/LoadingSpinner.module.css:2]: My thinking is to declare the CSS var here as the default since I remember coming into some issue leaving one undefined on the base selector. Seemed a little hat on a hat to re-declare it in a .md class but could see that making sense for clarity

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx
Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx:17]: Sure could break them into small subcomponents for cleanliness. In terms of rendering them out, I suppose I could do a ternary (though two conditions are a little ick to me), otherwise just a function that have a switch? i.e:

```
    {() => {
      switch (size) {
        case "sm":
          return <SmallSpinner />
        case "xs":
          return <ExtraSmallSpinner />
        case "md":
          return <MediumSpinner />
        default:
          return null
      }
    }}
    {size === "sm" ? (
      <SmallSpinner />
    ) : size === "xs" ? (
      <ExtraSmallSpinner />
    ) : (
      <MediumSpinner />
    )}
```

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx
Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx:108]: Likely, I did a quick scan and the [closest usages](https://github.com/search?q=org%3Acultureamp+%22loading-spinner+NOT+repo%3Acultureamp%2Fkaizen-design-system+NOT+repo%3Acultureamp%2Fkaizen-legacy&type=code) I found don't seem like they should actually be impacted

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx
Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx:17]: I suppose the ternary isn't too bad, a less code needed and we likely won't need any more than 3 sizes (likely)

Comment (HeartSquared) [reply packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx:17]: Or something like this :)

```suggestion
const Spinner = (size: LoadingSpinnerSize): JSX.Element => {
  if (size === "xs") return <ExtraSmallSpinner />
  if (size === "sm") return <SmallSpinner />
  return <MediumSpinner />
}
```

Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx:17]: Right, I getcha 😉

File: packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx
Comment (HeartSquared) [reply packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx:30]: Hmm...not really necessary imo to completely match the design examples unless our API is strictly limited to their selection (eg. variants). This story is showcasing the `classNameOverride` prop, and actually in this case, you're preventing the controls from working 😅

File: packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx
Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx]: I've moved all the spinners into a `SpinnerIcon` subcomponent, it was starting to look a little bloated and messy in the top of the component

Comment (HeartSquared) [reply packages/components/src/Loading/LoadingSpinner/LoadingSpinner.tsx]: As long as you don't export it at the entrypoint then it should be fine :) Because currently `SpinnerIcon` exists so it would conflict 😬

File: packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx
Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx:30]: Isn't this story more to showcase in our documentation, ie:

Screenshot 2024-10-10 at 11 35 15 AM

To our consumer the controls never did anything (at least on the docs page).

I suppose the question I'm wondering is, are we meant to have standard spinner colors, since they have been specifically designed?

Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx:30]: Maybe a separate story to delineate the difference between just classNameOverride and colors if they're needed? My main concern stems from have mild color variances in spinners

Comment (HeartSquared) [reply packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx:30]: I'd say that standard colours aren't needed. Using `classNameOverride` is the correct pattern.

The docs page doesn't show the controls for that story, but the individual story does :)

But yeah, I guess looking at that screenshot, I suppose showcasing a few more is nice 🤔 Perhaps change the order so that the one that can be changed by controls is the first one? Or move the spread args to the end so they can all be overridden by the controls?

That being said, color isn't the only use case for this (eg. margin, display), so I wonder if it would seem too specific and confusing only showing multiple colours 🤔 I'll leave the decision to you :)

Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx:30]: mmm I might check in with Jess to see how important those colors are to document, it could be a null issue in which case 🤷 but yeah to your point maybe leaving the classNameOverride story clean is good since it doesn't technically just to color

Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx:43]: Cool, so just chatted with Jess. They do have a set of colors that they use, but I said I wouldn't add scope to the ticket. I've create an additional story to be a little more explicit on the colors but kept it separate from the classNameOverride story

File: packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx
Comment (HeartSquared) [inline packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx:37]: Less lines if we revert 😉 
```suggestion
args: { classNameOverride: "text-green-400" },
```

Comment (mcwinter07) [reply packages/components/src/Loading/LoadingSpinner/_docs/LoadingSpinner.stories.tsx:37]: Now that's usin' the ol' noggin 👍  will do

**Review by HeartSquared — APPROVED** [review]
LGTM 🚀

**Comment by mcwinter07** [comment]
Holding off on merging this till Monday so we can align with design publishing Figma

**Pull request merged**
